native/src/main/java/io/ballerina/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -174,7 +174,7 @@ public static Object getAllRecords(BObject channel, int skipHeaders, BTypedesc t
                 }
                 Object[] out = outList.toArray();
                 return ValueCreator.createArrayValue(out, TypeCreator.createArrayType(describingType));
-            } else {
+            } else if (describingType.getTag() == TypeTags.ARRAY_TAG) {
```

Comment:
Here only the array type is checked not string[]. How does the function behave if I give a return type of boolean[]. If this is addressed it is fine.

Reply:
if we give the return type as boolean[] it returns the following error message `incompatible type for parameter 'returnType' with inferred typedesc value: expected 'typedesc<(string[]|map<anydata>)>', found 'typedesc<boolean>'`. this is handles in [here](https://github.com/sahanHe/module-ballerina-io/blob/e6b9e1aed7f24cc77568f0fb702efd18da79d752/ballerina/file_csv_io.bal#L28) as the expected type is `string[]|map<anydata>[]`

Reply:
Ack